app/src/apdu_handler.c

```diff
@@ -31,12 +31,51 @@
 #include "zxmacros.h"
 
 __Z_INLINE void handleGetPubkey(volatile uint32_t *flags, volatile uint32_t *tx, uint32_t rx) {
-    extractHDPath(rx, OFFSET_DATA);
+    if (rx != 6) {
+        THROW(APDU_CODE_DATA_INVALID);
+    }
 
     uint8_t requireConfirmation = G_io_apdu_buffer[OFFSET_P1];
+    const uint8_t slotIdx = G_io_apdu_buffer[OFFSET_DATA];
+
+    char buffer[20];
+    snprintf(buffer, sizeof(buffer), "%d", slotIdx);
+    zemu_log_stack(buffer);
+
+    account_slot_t slot;
+    zxerr_t err = slot_getSlot(slotIdx,(uint8_t *) &slot, sizeof(slot));
+    snprintf(buffer, sizeof(buffer), "err: %d", err);
+    zemu_log_stack(buffer);
+
+    if (err == zxerr_no_data) {
+        zemu_log_stack("Empty slot");
+        THROW(APDU_CODE_DATA_INVALID);
+    }
+
+    if (err != zxerr_ok) {
+        THROW(APDU_CODE_EXECUTION_ERROR);
+    }
+
+    //We want to reuse extractHDPath, as it works on G_io_apdu_buffer we have to follow 
+    MEMCPY(G_io_apdu_buffer, &slot.path, sizeof(slot.path));
+    rx = sizeof(slot.path);
+    extractHDPath(rx, 0); //this puts hdPath into hdPath global variable
```

Comment:
do we really need to have this in global variable?

Reply:
extractHDPath is in /common folder. I am somewhat reluctant to change it. And it requires that G_io_apdu_buffer is used. Or is this about rx?

Reply:
If this comment is about hdPath, the path needs to be in a global variable for displaying (unless a huge refactor of zondax library occours). What I believe is a bit ugly is that we take the path from G_io_apdu_buffer. But extractHDPath is used also in process_chunk function (part of sign transaction). But if you strongli believe that this refactor should be done I'll do it...

Reply:
I came to conculsion that using extractHDPath at this spot is incorrect. The derivation path was parsed and validated by slot_parseSlot. At this point a plain memccpy is adequate (we might want to make a duplicate validation). While examining this I found two places where path validation is duplicate. I will push a refactor soon.

Reply:
If the path was validated when parsing a slot, there is no need to make a duplicate validation.